packages/orbit-components/src/Itinerary/ItinerarySegment/ItinerarySegmentDetail/index.jsx

```diff
@@ -0,0 +1,209 @@
+  `}
+`;
+
+// $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198
+StyledExpandable.defaultProps = {
+  theme: themeDefault,
+};
+
+const StyledExpandableContent = styled.div`
+  ${({ $offset, theme }) => css`
+    padding: ${theme.orbit.spaceSmall};
+    position: relative;
+    z-index: 1;
+    margin-${left}: ${parseInt(theme.orbit.spaceXSmall, 10) + $offset}px;
+  `}
+`;
+
+const StyledHeadingOffset = styled.div`
+  ${({ theme }) => css`
+    margin-${left}: ${theme.orbit.spaceXLarge};
+  `}
+`;
+
+const StyledIcon = styled.div`
+  ${({ theme, isFirst, isLast }) => css`
+    display: flex;
+    align-items: center;
+    position: relative;
+    padding: ${theme.orbit.spaceXXSmall};
+    z-index: 3;
+    &:after {
+      content: "";
+      position: absolute;
+      width: 100%;
+      height: 200%;
+      left: 0;
+      background: ${theme.orbit.paletteWhite};
+      ${isFirst &&
+      css`
+        border-radius: ${theme.orbit.spaceLarge} ${theme.orbit.spaceLarge} 0 0;
+      `}
+      ${isLast &&
+      css`
+        border-radius: 0 0 ${theme.orbit.spaceLarge} ${theme.orbit.spaceLarge};
+      `}
+      z-index: -1;
+    }
+  `}
+`;
+
+// $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198
+StyledIcon.defaultProps = {
+  theme: themeDefault,
+};
+
+// $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198
+StyledExpandableContent.defaultProps = {
+  theme: themeDefault,
+};
+
+const ItinerarySegmentDetail = ({ duration, summary, content, icon }: Props): React.Node => {
+  const { opened, toggleOpened } = usePart();
+  const { calculatedWidth } = useWidth();
+  const [{ height: slideHeight }, slideRef] = useBoundingRect({ height: opened ? null : 0 });
+  const randomId = useRandomIdSeed();
+
+  return (
+    <StyledWrapper opened={opened}>
+      <StyledInnerWrapper>
+        <Stack align="center" spacing="small">
+          <StyledDuration $minWidth={calculatedWidth || 60}>
+            <Text size="small">{duration}</Text>
+          </StyledDuration>
+          <StyledDetailsIcon>
+            <ItineraryIcon isDetails>{icon}</ItineraryIcon>
+          </StyledDetailsIcon>
+          <Stack justify="center" shrink direction="column" spacing={opened ? "medium" : "none"}>
+            <StyledSummary opened={opened}>{summary}</StyledSummary>
+          </Stack>
+          {opened ? <ChevronUp /> : <ChevronDown />}
+        </Stack>
+      </StyledInnerWrapper>
+      <Slide
+        maxHeight={slideHeight}
+        expanded={opened}
+        id={randomId("slide")}
+        ariaLabelledBy={randomId("slide")}
+      >
+        <StyledExpandable ref={slideRef} onClick={toggleOpened}>
+          <StyledExpandableContent $offset={calculatedWidth}>
+            {content &&
+              content.map(({ title, items }, idx) => {
+                return (
+                  <React.Fragment key={randomId(String(idx))}>
```

Comment:
Scratch what I said about avoiding index. In this case it would be really hard to improve render performance, so we should think about this separately as we reevaluate our components as a separate task. But you can use `idx` directly, you don't need to use `randomId`, `key` only needs to be unique for siblings.